### DISCUSSION TITLE
How to define Build & Development Settings when deploy with cli?

### DISCUSSION THREAD

**Discussion by winston0410** [vercel/community#692] [Help] [answered]
When I deploy application to Vercel with the cli, is it possible to define Build & Development Settings, which I can find in the web application? Those includes **build command**, **output directory** and **install command**

**Comment by swarnava** [comment, marked as answer]
When you deploy the project, CLI should prompt you to define/override the settings:

<img width="512" alt="image" src="https://user-images.githubusercontent.com/1162991/176987641-649543a3-7a16-4c50-9013-776e87b60767.png">

**Reply by winston0410** [reply]
But how can I define install command here? And if a local configuration is not used to defined all these variables, how can I deploy it with CICD like CircleCI?

**Reply by winston0410** [reply]
<img width="883" alt="Screenshot 2022-07-02 at 10 20 33 AM" src="https://user-images.githubusercontent.com/59228178/176994597-ac2b9726-9f90-47cb-a3b9-8dd8e1b0faff.png">

**Reply by swarnava** [reply]
it supports `&&` operator like bash, so you can override `Build command` with `pnpm install && pnpm rebuild` or something as you like

**Reply by winston0410** [reply]
> it supports `&&` operator like bash, so you can override `Build command` with `pnpm install && pnpm rebuild` or something as you like

Thanks for your reply! so does that mean that I should handle install and build in the same config there?

Also for the CICD issue, what can I do with that?

**Comment by Ethan-Arrowood** [comment]
Hello @winston0410 ! I have some great news, soon we are releasing the ability to specify some project settings via `vercel.json` enabling you to have your settings configured just as you'd like them for your first deployment from the CLI. Before we release this feature I'm looking to gain some user feedback. If you'd be interested in trying it out please reach out to me at ethan.arrowood@vercel.com and we can find a time to collaborate. Thank you!
